### PR TITLE
Simplify momentum initialization in example input files

### PR DESCRIPTION
**Pull request by NeilZaim**
Very often, we use in the example input file `particles.momentum_distribution_type= "Gaussian"` when the particles are initialized at rest or with constant momentum. Most of the time, this is done using the (implicit) fact that the mean and thermal velocities are all 0 by default. While, this is not a serious issue, it would be more convenient to see immediately that the particles are initialized with constant momentum or at rest, without having to look if `particles.ux_m` or `particles.ux_th` are defined someplace else in the input file.

So in this PR, I tried to make the initialization of the momentum in our example input files more explicit. More specifically, I have made the following changes:
- I've added a new `momentum_distribution_type`, called `at_rest` (let me know if you'd prefer another name), which just initializes all the momentum components to 0, and I've used it whenever appropriate in the input file.
- Whenever a constant but nonzero initial momentum is used, I've replaced `gaussian` with `constant` in the input file.
- In the documentation, it was written that the parameters determining the initial momentum distributions (`ux_m`, `ux_th`, etc.) were required, which is not the case. I've changed that and added the default values.
- Somewhat unrelated but I've added some `_rt` in the `PlasmaInjector` class.

**Edit**: looks like some benchmarks have changed after this PR. In every case, it looks like it is coming from changing the injection type from `gaussian` to `at_rest`, which leads to fewer calls to random number generation, and is (I think) equivalent to changing the random seed for subsequent processes that need random numbers (all the tests that need a benchmark reset have random number generation somewhere else, either through qed or through `gaussian` momentum distribution that remain for some species). So I think that these benchmarks can be safely updated.

### CODE REVIEW

**Comment by NeilZaim** [comment]
It looks like the test `BTD_ReducedSliceDiag` has crashed, I'm not sure why (it didn't crash before merging with development). I will merge again and see if it crashes again.

Edit: After restarting CI, the test worked, so I'm not sure what happened.

File: Source/Initialization/PlasmaInjector.cpp
Comment (EZoni) [inline Source/Initialization/PlasmaInjector.cpp]:
```suggestion
        constexpr amrex::Real ux = 0._rt;
        constexpr amrex::Real uy = 0._rt;
        constexpr amrex::Real uz = 0._rt;
```

File: Source/Initialization/PlasmaInjector.cpp
Comment (EZoni) [inline Source/Initialization/PlasmaInjector.cpp]:
```suggestion
        amrex::Real ux = 0._rt;
        amrex::Real uy = 0._rt;
        amrex::Real uz = 0._rt;
```

File: Source/Initialization/PlasmaInjector.cpp
Comment (EZoni) [inline Source/Initialization/PlasmaInjector.cpp]:
```suggestion
        amrex::Real ux_m = 0._rt;
        amrex::Real uy_m = 0._rt;
        amrex::Real uz_m = 0._rt;
        amrex::Real ux_th = 0._rt;
        amrex::Real uy_th = 0._rt;
        amrex::Real uz_th = 0._rt;
```

File: Source/Initialization/PlasmaInjector.cpp
Comment (EZoni) [inline Source/Initialization/PlasmaInjector.cpp]:
```suggestion
        amrex::Real ux_m = 0._rt;
        amrex::Real uy_m = 0._rt;
        amrex::Real uz_m = 0._rt;
        amrex::Real ux_th = 0._rt;
        amrex::Real uy_th = 0._rt;
        amrex::Real uz_th = 0._rt;
```

File: Source/Initialization/PlasmaInjector.cpp
Comment (EZoni) [inline Source/Initialization/PlasmaInjector.cpp]:
```suggestion
        amrex::Real beta = 0._rt;
        amrex::Real theta = 10._rt;
```

File: Source/Initialization/PlasmaInjector.cpp
Comment (EZoni) [inline Source/Initialization/PlasmaInjector.cpp]:
```suggestion
        amrex::Real beta = 0._rt;
        amrex::Real theta = 10._rt;
```

File: Source/Initialization/PlasmaInjector.cpp
Comment (EZoni) [inline Source/Initialization/PlasmaInjector.cpp]:
```suggestion
        amrex::Real u_over_r = 0._rt;
```

**Review by EZoni — CHANGES_REQUESTED** [review]
Thank you for this PR, Neil! I think it's a great idea to make our inputs more self-explanatory. I have only two comments:

1. Regarding the name `at_rest`, I think I'm fine with it, but some might think that having an "underscored" name in the input could be a bit annoying. It might be worth thinking of single-word alternatives, such as `rest`, `zero`, `null`, and similar ones. I'll let other developers chime in on this.
2. I added a few suggestions to include the prefix `amrex::` in front of the `Real` types used in the code here. It is a small stylistic detail, but I think we decided some time ago that all AMReX types should be prefixed with `amrex::` in WarpX (in fact we wrote it even in our contributing guidelines), so I would suggest to include that prefix in any new code.

**Comment by NeilZaim** [comment]
Hi @EZoni and thanks for the review.

Regarding 1., I personally like `at_rest` because it's short and very explicit. But I'm fine with changing this if needed. What do you think is wrong with having underscores? We already have some for this specific input parameter (`maxwell_boltzmann` or `maxwell_juttner`) and for a few other input parameters and we also abundantly use it for the names of the input parameters themselves (e.g. `<species_name>.momentum_distribution_type` in this case). We could consider ignoring the underscores for robustness, which would be easy to do, but it'd probably be a good idea to do this consistently for all input parameters that read a string.

Regarding 2., I agree with you. Usually, I try to always add the `amrex::` prefix, unless when I do small modifications on a file that does not use this convention (mainly for consistency within a file, and also due to laziness). But in this case I modified nearly all the `Real` variables in the file, so it definitely makes sense to add the prefix. I've added it in a few other places in this file.
Edit: in the end I went ahead and removed the `using namespace amrex` in this file, adding all the necessary `amrex::` prefixes.

**Comment by EZoni** [comment]
Thank you for adding the `amrex::` prefix consistently, Neil! I think the reason why CI is failing is that there are still a few missing, when we compile with openPMD support. The diff below should tell you which ones and it should fix CI:
```
--- a/Source/Initialization/PlasmaInjector.cpp
+++ b/Source/Initialization/PlasmaInjector.cpp
@@ -310,7 +310,7 @@ PlasmaInjector::PlasmaInjector (int ispecies, const std::string& name)
         queryWithParser(pp_species_name, "z_shift",z_shift);
 
 #ifdef WARPX_USE_OPENPMD
-        if (ParallelDescriptor::IOProcessor()) {
+        if (amrex::ParallelDescriptor::IOProcessor()) {
             m_openpmd_input_series = std::make_unique<openPMD::Series>(
                 str_injection_file, openPMD::Access::READ_ONLY);
 
@@ -375,11 +375,11 @@ PlasmaInjector::PlasmaInjector (int ispecies, const std::string& name)
 
         // Broadcast charge and mass to non-IO processors
         if (!charge_is_specified && !species_is_specified)
-            ParallelDescriptor::Bcast(&charge, 1,
-                ParallelDescriptor::IOProcessorNumber());
+            amrex::ParallelDescriptor::Bcast(&charge, 1,
+                amrex::ParallelDescriptor::IOProcessorNumber());
         if (!mass_is_specified && !species_is_specified)
-            ParallelDescriptor::Bcast(&mass, 1,
-                ParallelDescriptor::IOProcessorNumber());
+            amrex::ParallelDescriptor::Bcast(&mass, 1,
+                amrex::ParallelDescriptor::IOProcessorNumber());
 #else
         amrex::Abort("Plasma injection via external_file requires openPMD support: "
                      "Add USE_OPENPMD=TRUE when compiling WarpX.\n");
```

Regarding the name `at_rest`, I'm totally fine with it. I don't think there's anything wrong with having underscores, I just remember some Zoom discussions in the past where it seemed that some developers/users could have a preference for single-word names over anything else essentially. But I'm personally totally fine with `at_rest`, it's short and self-explanatory.

**Comment by NeilZaim** [comment]
Thanks Edoardo for simplifying my work for the OpenPMD compilation. :) Looks like it's working now.

**Review by EZoni — APPROVED** [review]
Thank you again, Neil! This looks good to me. I can merge it after we have at least one more review, since this is changing the input syntax among other things.

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
Thank you Neil, LGTM! :)

Also thank you Edoardo for the review!

**Pull request merged**
